File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const obj = {};
    for(let i = 0; i < participant.length; i++){
        if(obj[participant[i]] === undefined) {
            obj[participant[i]] = 1;
        } else {
            obj[participant[i]] += 1;
        }
    }
    for(let i = 0; i < completion.length; i++){
        obj[completion[i]] -= 1;
    }
    for(key in obj){
        if(obj[key] > 0) return key
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dict = { }
    for cp in completion:
        if cp in dict:
            dict[cp] += 1
        else:
            dict[cp] = 1
    
    for p in participant:
        if p in dict:
            if dict[p] == 1:
                del dict[p]
            else:
                dict[p] -= 1
        else:
             return p   